### PR TITLE
Add initial support for Journey mode in SSC.

### LINKED ISSUE #1907: Journey mode SSC reserach not persisted
In journey mode with SSC, the client expects the server to send the proper NetModule back down when it researches something. This is not implemented in vanilla, and we need to implement it ourselves. The client will instead send a research request NetModule and that correlates with intent to research -- so we need to acknowledge it and send back the right thing.

### PR DESCRIPTION
**Pull request by Olink**
Fixes #1907

### CODE REVIEW

**Review by hakusaro — APPROVED** [review]

File: TShockAPI/DB/ResearchDatastore.cs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+			if (!(itemsSacrificed.ContainsKey(itemId)))
+				itemsSacrificed[itemId] = 0;
+
+			var sql = @"insert into Research (WorldId, PlayerId, ItemId, AmountSacrificed, TimeSacrificed) values (@0, @1, @2, @3, @4)";
+			var result = database.Query(sql, Main.worldID, player.Account.ID, itemId, amount, DateTime.Now);
+			if (result == 1)
+			{
+				itemsSacrificed[itemId] += amount;
+			}
+
```
Comment (QuiCM) [inline TShockAPI/DB/ResearchDatastore.cs:134]: Should we have a logging message should the query fail?

Comment (hakusaro) [reply TShockAPI/DB/ResearchDatastore.cs:134]: mo

Comment (hakusaro) [reply TShockAPI/DB/ResearchDatastore.cs:134]: mo*

Comment (hakusaro) [reply TShockAPI/DB/ResearchDatastore.cs:134]: no**

Comment (Olink) [reply TShockAPI/DB/ResearchDatastore.cs:134]: See latest commit.

File: TShockAPI/DB/ResearchDatastore.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+				db.GetSqlType() == SqlType.Sqlite
+					? (IQueryBuilder)new SqliteQueryCreator()
+					: new MysqlQueryCreator());
+			try
+			{
+				creator.EnsureTableStructure(table);
+			}
+			catch (DllNotFoundException)
+			{
+				System.Console.WriteLine("Possible problem with your database - is Sqlite3.dll present?");
```
Comment (hakusaro) [inline TShockAPI/DB/ResearchDatastore.cs]:
```suggestion
				Console.WriteLine("Possible problem with your database - is Sqlite3.dll present?");
```

**Review by hakusaro — APPROVED** [review]

**Review by QuiCM — APPROVED** [review]

**Pull request merged**